### DISCUSSION TITLE
Request for a requirements.txt file

### DISCUSSION THREAD

**Discussion by itazap** [upkie/upkie#466] [General]
## Bug description

This library depends on specific versions of certain packages, and it would really help streamline the installation process by contributing a requirements.txt file to outline the necessary dependent packages. Thanks!

## Expected behavior

users can then run `pip install -r requirements.txt`

### Code

this file can simply be created with: 
`pip freeze > requirements.txt`

**Comment by stephane-caron** [comment]
Thank you for raising this. While `requirements.txt` files are still a thing, in this project we rather specify dependencies in `pyproject.toml`:

https://github.com/upkie/upkie/blob/96d2fded4480e656e89131e5e31402bb24f9d49f/pyproject.toml#L30-L37

If your use case is to install the project from source, you can run from the repository:

```console
pip install .
```

This will automatically fetch all dependencies from PyPI with proper versions, and install any local changes you may have from source.

**Reply by itazap** [reply]
Thank you @stephane-caron ! I was having conflicts with the dependencies for ppo_balancer, but a fresh restart helped. On that note, does https://github.com/upkie/ppo_balancer require dependencies like xcode? Unable to get the make train to work without errors. Thanks again for your response!

Update: unable to get passed this error:

`ERROR: /private/var/tmp/_bazel_itazaporozhets/498dbf992c37892e7013f5ba2c923d0f/external/local_config_cc/BUILD:48:19: in cc_toolchain_suite rule @local_config_cc//:toolchain: cc_toolchain_suite '@local_config_cc//:toolchain' does not contain a toolchain for cpu 'darwin_x86_74'
`

was wondering if there is a cpp-related or intel-related dependency ? I was able to run `make test_policy` but not `make train`. Thanks.

**Reply by stephane-caron** [reply]
Hmm the string "darwin_x86_74" should be "darwin_x86_64" (standing for 64 bits). I don't find the incorrect string in the repositories, so it is possible you have some local changes. You can search for that string in your local files and correct it if found, or maybe start afresh from a clean `git clone` of the repositories?

For info, the string "darwin_x86_64" appears in the [`.bazelrc`](https://github.com/upkie/upkie/blob/b5af2bef92673dcf840bc7c8876a923faa725b4a/.bazelrc) file of the upkie repository.

**Reply by itazap** [reply]
Acutally yes the error was with "64", that's my mistake, I clipped the error when copy-pasting and manually retyping the error
